core/monitor/LogtailMetric.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+#include <string>
+#include <atomic>
+#include "common/Lock.h"
+#include "log_pb/sls_logs.pb.h"
+
+
+namespace logtail {
+
+
+class BaseMetric {
+    protected:
+        std::string mName;
+        std::atomic_long mVal;
+    public:
+        const uint64_t GetValue() const;
+        const std::string GetName() const;
+        virtual BaseMetric* CopyAndReset() = 0;
+};
+
+
+class Gauge : public BaseMetric {
+    public:
+        Gauge(const std::string name);   
+        void Set(uint64_t val);
+        Gauge* CopyAndReset();
+};
+
+
+class Counter : public BaseMetric {
+    public:
+        Counter(const std::string name);
+        void Add(uint64_t val);
+        Counter* CopyAndReset();
+};
+
+using MetricPtr = std::shared_ptr<BaseMetric>;
+using CounterPtr = std::shared_ptr<Counter>;
+using GaugePtr = std::shared_ptr<Gauge>;
```

Comment:
CounterPtr和GaugePtr有什么用？用指针是因为基类指针能转换，子类指针并没有什么用
```suggestion
using MetricPtr = std::shared_ptr<BaseMetric>;
```